### ISSUE TITLE
[types] wrong type definition when using `Date` as source primitive type in `Mapper#addTypeConverter`

### ISSUE THREAD

**Issue by micalevisk** [nartc/mapper#368]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

regarding typescript types

```
mapper.addTypeConverter(String, Number, (str) => 123)
#                                        ^^^ string ✅
mapper.addTypeConverter(Date, Number, (date) => 123)
#                                      ^^^ string ❌
```

the type of `date` parameter above is `string` instead of `Date`

### Expected behavior

the type definition of `date` parameter below must be `Date`

```
mapper.addTypeConverter(Date, Number, (date) => 123)
```

### Screenshots

![image](https://user-images.githubusercontent.com/13461315/139161429-4c842d24-ede6-41b9-bc4a-b4d02bd46f8b.png)

### Package

- [ ] I don't know.
- [ ] <code>@automapper/core</code>
- [ ] <code>@automapper/classes</code>
- [ ] <code>@automapper/nestjs</code>
- [ ] <code>@automapper/pojos</code>
- [ ] <code>@automapper/sequelize</code>
- [x] <code>@automapper/types</code>
- [ ] Other (see below)

### AutoMapper version

6.3.0

**Comment by micalevisk** [comment]
it worked.

---

@nartc side-suggestion that I think you are aware: when you write a commit that is related to some issue, can you please link that commit with the issue, using some of these GH keywords https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword? Sometimes I find myself curious to know what changes you've made to solve the issue :smile: linking the commit with the issue will help me find out this easily 

Doing something like `git commit -m 'foo' -m 'closes #368'` (btw this will make the issue closes automatically)

**Comment by nartc** [comment]
Nice!

Yeah I am aware. The Commitizen flow should have that question asked (related issue) but sometimes I just commit using: `git commit` directly and forget to put in the issue (mainly because I am lazy to look up the issue number 😅). I’ll remember to do it next time. Thanks for the suggestion.

If you’re still curious about the changes I made, CHANGELOG is a good place to explore (though I do admit that I sneak in changes into the commits far too often 🥲)
